crates/aptos/src/move_tool/bytecode.rs

```diff
@@ -0,0 +1,306 @@
+};
+use move_coverage::coverage_map::CoverageMap;
+use move_disassembler::disassembler::{Disassembler, DisassemblerOptions};
+use move_ir_types::location::Spanned;
+use std::{
+    fs,
+    path::{Path, PathBuf},
+    process::Command,
+};
+
+const REVELA_EXE_ENV: &str = "REVELA_EXE";
+#[cfg(target_os = "windows")]
+const REVELA_EXE: &str = "revela.exe";
+#[cfg(not(target_os = "windows"))]
+const REVELA_EXE: &str = "revela";
+
+const DISASSEMBLER_EXTENSION: &str = "mv.asm";
+const DECOMPILER_EXTENSION: &str = "mv.move";
+
+/// Disassemble the Move bytecode pointed to in the textual representation
+/// of Move bytecode.
+///
+/// For example, if you want to disassemble an on-chain package `PackName` at account `0x42`:
+/// 1. Download the package with `aptos move download --account 0x42 --package PackName --bytecode`
+/// 2. Disassemble the package bytecode with `aptos disassemble --package-path PackName/bytecode_modules`
+#[derive(Debug, Parser)]
+pub struct Disassemble {
+    #[clap(flatten)]
+    pub command: BytecodeCommand,
+}
+
+/// Decompile the Move bytecode pointed to into Move source code.
+///
+/// For example, if you want to decompile an on-chain package `PackName` at account `0x42`:
+/// 1. Download the package with `aptos move download --account 0x42 --package PackName --bytecode`
+/// 2. Decompile the package bytecode with `aptos decompile --package-path PackName/bytecode_modules`
+#[derive(Debug, Parser)]
+pub struct Decompile {
+    #[clap(flatten)]
+    pub command: BytecodeCommand,
+}
+
+#[derive(Debug, Clone, Copy, Default)]
+enum BytecodeCommandType {
+    #[default]
+    Disassemble,
+    Decompile,
+}
+
+#[derive(Debug, Parser)]
+pub struct BytecodeCommand {
+    /// Set programmatically
+    #[clap(skip)]
+    command_type: BytecodeCommandType,
+
+    /// Treat input file as a script (default is to treat file as a module)
+    #[clap(long)]
+    pub is_script: bool,
+
+    /// The path to a directory containing Move bytecode files with the extension `.mv`.
+    /// The tool will process all files find in this directory
+    ///
+    /// If present, a source map at the same location ending in `.mvsm` and the source
+    /// file itself ending in`.move` will be processed by the tool.
+    #[clap(long)]
+    pub package_path: Option<PathBuf>,
+
+    /// Alternatively to a package path, path to a single bytecode file which should be processed.
+    #[clap(long)]
+    pub bytecode_path: Option<PathBuf>,
+
+    /// (Optional) Currently only for disassemble: path to a coverage file for the VM in order
+    /// to print trace information in the disassembled output.
+    #[clap(long)]
+    pub code_coverage_path: Option<PathBuf>,
+
+    /// Output directory for the generated file. Defaults to the directory of the
+    /// `path/module.mv` file if not provided. The disassembled output is stored
+    /// at `output_dir/module.mv.asm`, and decompiled output at
+    /// `output_dir/module.mv.move`.
+    #[clap(long, value_parser)]
```

Comment:
I forget if we need the "value_parser" for all paths, the parser became really confusing these last few upgrades.  We'll make a patch later if it needs it for all of them.